reporter/diff.js

```diff
@@ -907,7 +907,7 @@ QUnit.diff = ( function() {
 		// Performance analysis: https://neil.fraser.name/news/2010/11/04/
 		best = 0;
 		length = 1;
-		while ( true ) {
+		for ( ;; ) {
```

Comment:
Which rule is this change accomodating? I think the `while ( true )` form is much easier to read than the `for ( ;; )`

Reply:
[no-constant-condition](https://eslint.org/docs/rules/no-constant-condition) disallows the `while (true)` form, but we can allow it by setting `"checkLoops": false`.

Reply:
Agreed, we should allow constant expression in loops. I imagine `for ( ;; ) {` might be faster in some sense in some engines, maybe, but not worth the cost of code readability here. If this was a really hot code path that would get faster from this, then it'd be worth considering (but not without a clarifying inline comment in that case).